packages/cryptoplease/lib/features/popular_tokens/popular_token_list.dart

```diff
@@ -1,55 +1,87 @@
 import 'package:auto_route/auto_route.dart';
-import 'package:dfunc/dfunc.dart';
+import 'package:decimal/decimal.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 import '../../../../../core/amount.dart';
-import '../../../../../core/conversion_rates/context_ext.dart';
 import '../../../../../core/presentation/format_amount.dart';
 import '../../../../../core/tokens/token.dart';
 import '../../../../../core/user_preferences.dart';
 import '../../../../../l10n/device_locale.dart';
 import '../../../../../routes.gr.dart';
 import '../../../../../ui/colors.dart';
 import '../../../../../ui/token_icon.dart';
+import '../../di.dart';
+import '../../ui/loader.dart';
+import 'src/market_bloc.dart';
 
-final _popularTokenList = <Token>[
+final _defaultTokens = [
   Token.sol,
+  Token.usdc,
 ];
 
 class PopularTokenList extends StatelessWidget {
   const PopularTokenList({super.key});
 
   @override
-  Widget build(BuildContext context) => SliverList(
-        delegate: SliverChildListDelegate(
-          _popularTokenList.map(_TokenItem.new).toList(),
+  Widget build(BuildContext context) => BlocProvider(
+        create: (context) => sl<MarketBloc>(
+          param1: context.read<UserPreferences>().fiatCurrency,
+        )..add(const MarketEventFetch()),
+        child: BlocBuilder<MarketBloc, MarketDetailsState>(
+          builder: (context, state) {
+            const loader = SliverToBoxAdapter(
+              child: SizedBox(
+                height: 80,
+                child: LoadingIndicator(),
+              ),
+            );
+
+            return state.when(
+              initial: () => loader,
+              processing: () => loader,
+              failure: (_) => SliverList(
+                delegate: SliverChildListDelegate(
+                  _defaultTokens.map((e) => _TokenItem(e, 0.0)).toList(),
```

Comment:
I don't think it makes sense to diplay stub tokens with `0` amount in case of a failure, it would make more sense to show the error instead.

Also, we really need to cache this info locally, but this can be done in another PR.